src/sql/parts/jobManagement/common/agentJobCacheService.ts

```diff
@@ -0,0 +1,54 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+'use strict';
+
+import { Injectable } from '@angular/core';
+import * as sqlops from 'sqlops';
+import * as vscode from 'vscode';
+import { IAgentJobCacheService } from './interfaces';
+
+@Injectable()
+export class AgentJobCacheService implements IAgentJobCacheService {
```

Comment:
Does this need to be a service?  Or can it just be a normal class that the JobService uses?

Reply:
Also, let's not put agent in the class names unless it is a component that will only work with agent.  I think in this case this is part of the generic job support.